### PR TITLE
Bug 1274201 - Fixing sudo non-interactive test

### PR DESCRIPTION
**Pull request by brenton**
https://bugzilla.redhat.com/show_bug.cgi?id=1274201#c13

### CODE REVIEW

**Comment by brenton** [comment]
@smunilla PTAL!

**Comment by smunilla** [comment]
LGTM

**Comment by brenton** [comment]
@smunilla, @dgoodwin, let me know if my updates make sense to you guys.  I think the git commit messages explain everything.

**Comment by dgoodwin** [comment]
Looks good to me, looks like a minor line length complaint from pylint.

If time allowed and you wanted, it probably wouldn't be too hard to add a test, we've got some stuff verifying inventory already in there. (utils/test/cli_installer_tests.py)

File: utils/src/ooinstall/cli_installer.py [hunk 1]
```diff
@@ -339,7 +339,10 @@ def get_hosts_to_run_on(oo_cfg, callback_facts, unattended, force):
                 # new nodes.
             elif host.node:
                 click.echo("{} is already an OpenShift Node".format(host))
-                hosts_to_run_on.remove(host)
+                # force is only used for reinstalls so we don't want to remove
+                # anything.
+                if not force:
+                    hosts_to_run_on.remove(host)
```
Comment (brenton) [inline utils/src/ooinstall/cli_installer.py]: @smunilla, without this the following installer config wasn't working with `--force`:

~~~
ansible_log_path: /tmp/ansible.log
ansible_ssh_user: root
hosts:
- hostname: ose3-master.example.com
  ip: 192.168.133.2
  master: true
  node: true
  public_hostname: ose3-master.example.com
  public_ip: 192.168.133.2
- hostname: ose3-node1.example.com
  ip: 192.168.133.3
  node: true
  public_hostname: ose3-node1.example.com
  public_ip: 192.168.133.3
variant: openshift-enterprise
variant_version: '3.1'
version: v1
~~~

**Comment by brenton** [comment]
pylint is reporting too many branches.  I definitely agree.  I walked @dgoodwin through the change and right now the plan is to at least get this in from of QE and then tomorrow write unit tests for this.  I just wasted familiar enough with the rest of the code to do that first. :(

**Pull request merged**